### ISSUE TITLE
`!VALUE(x)` in a method signature is not parsed, and the method is lost

### ISSUE THREAD

**Issue by oisee** [abaplint/abaplint#4308]
abaplint parses `!x` and parses `VALUE(x)`, but not the two together. A
parameter written `!VALUE(x)` makes the `METHODS` / `CLASS-METHODS` statement
come back as `Unknown`, so that method is missing from
`getClassDefinition().methods`. Other methods of the class are unaffected.

In practice the form is generated rather than typed, so it covers every
parameter of every method in a class, and what arrives is a class with almost
no signatures. In two classes read off a system, abaplint parses 2 of 15
methods and 1 of 9; removing the `!` before parsing gives all of them.

`findIssues()` does report a `parser_error`, but it points at the `CLASS`
token rather than at the parameter, and a caller using the
`getClassDefinition()` API sees only a missing method.

### Reproducer

```js
const {MemoryFile, Registry} = require("@abaplint/core");
const source = `CLASS c DEFINITION PUBLIC.
  PUBLIC SECTION.
    CLASS-METHODS m1 IMPORTING !VALUE(iv_x) TYPE i.
ENDCLASS.
CLASS c IMPLEMENTATION.
ENDCLASS.`;
const reg = new Registry().addFile(new MemoryFile("c.clas.abap", source)).parse();
const obj = reg.getFirstObject();
obj.getABAPFiles()[0].getStatements().map(s => s.get().constructor.name);
// [ 'ClassDefinition', 'Public', 'Unknown', 'EndClass', 'ClassImplementation', 'EndClass' ]
obj.getClassDefinition().methods;   // []
```

### The forms, measured separately

| declaration | result |
| --- | --- |
| `IMPORTING iv_x TYPE i` | parsed |
| `IMPORTING !iv_x TYPE i` | parsed |
| `IMPORTING VALUE(iv_x) TYPE i` | parsed |
| `IMPORTING !VALUE(iv_x) TYPE i` | **`Unknown`**, method missing |
| `IMPORTING !REFERENCE(iv_x) TYPE i` | **`Unknown`**, method missing |

Lower case `!value(...)` behaves the same as the upper-case form.

Version: `@abaplint/core` 2.120.55, current on npm at the time of writing.

### Where I hit it

I read AMDP method bodies out of classes exported from a system and need the
signature to know which names in the body are IN table parameters. With this
form the signature is missing, so a body referring to `:it_ddls` — a
parameter declared three lines above it — looks like a body reading an
undeclared table variable. The message points somewhere other than the
defect, which is what made it expensive to find.

I work around it by removing the `!` before `VALUE(` prior to parsing. The
`!` escapes the identifier so it is not read as a keyword; it carries no
meaning for the interface, so removing it before parsing changes nothing
about what is read.

### Where it seems to come from

Read off the built package rather than the source, so treat it as a pointer
and not a diagnosis. The lexer emits `!VALUE` as a single `Identifier` token,
while `MethodParam` matches `VALUE` / `REFERENCE` as literal strings:

```js
// 2_statements/expressions/method_param.js
const ref   = seq("REFERENCE", tok(ParenLeft), Expressions.MethodParamName, tok(ParenRightW));
const value = seq("VALUE",     tok(ParenLeft), Expressions.MethodParamName, tok(ParenRightW));
```

and only `MethodParamName` tolerates a leading `!` — its regex is
`^!?\w*(\/\w+\/)?\w+$`. That is why `!iv_x` works and `!VALUE(iv_x)` does
not. Whether the escape belongs in those two alternatives or somewhere
earlier is yours to say; happy to send a PR if you point at the shape you
want.

Related: #2529, the same escape in front of a builtin function.

**Comment by larshp** [comment]
yea... recommend removing the exclamation marks

but also PRs welcome
